### PR TITLE
[NOBUILD][PyROOT exp] Port RDataFrame.AsNumpy

### PR DESCRIPTION
**Pull request by stwunsch**

### CODE REVIEW

**Comment by stwunsch** [comment]
@phsft-bot build with `-Dpyroot_experimental=ON`

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    # Cast input node to base RNode type
+    # TODO: Can this be removed if Take works now?
+    #df_rnode = cppyy.gbl.ROOT.RDF.AsRNode(df)
+
+    # Register Take action for each column
+    result_ptrs = {}
+    for column in columns:
+        #column_type = df_rnode.GetColumnType(column)
+        column_type = df.GetColumnType(column)
+        # TODO: Does this work with only Take?
```
Comment (etejedor) [inline bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py]: Do we still need this comment here?

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    if not columns:
+        columns = [c for c in df.GetColumnNames()]
+
+    # Exclude the specified columns
+    if exclude == None:
+        exclude = []
+    columns = [col for col in columns if not col in exclude]
+
+    # Cast input node to base RNode type
+    # TODO: Can this be removed if Take works now?
```
Comment (etejedor) [inline bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py]: And this?

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# Author: Stefan Wunsch CERN  08/2018
```
Comment (etejedor) [inline bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py]: 02/2019 :)

File: bindings/pyroot_experimental/PyROOT/CMakeLists.txt [hunk 4]
```diff
@@ -16,6 +16,7 @@ set(py_sources
   ROOT/pythonization/_tfile.py
   ROOT/pythonization/_tseqcollection.py
   ROOT/pythonization/_ttree.py
+  ROOT/pythonization/_rdataframe.py
```
Comment (etejedor) [inline bindings/pyroot_experimental/PyROOT/CMakeLists.txt]: Being picky: can we keep it alphabetical?

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py (hunk 1 above)
Comment (stwunsch) [reply bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py]: Ah damn, I forgot to remove them :D Yes, it works now with `Take` and we do not need the `RNode` cast anymore.

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py (hunk 2 above)
Comment (stwunsch) [reply bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rdataframe.py]: Sry for the messy PR :see_no_evil: I've cleaned it up.

**Comment by stwunsch** [comment]
Codacity is super great! :) @etejedor I've fixed the codacity warnings and your comments.

File: bindings/pyroot_experimental/PyROOT/test/rdataframe_asnumpy.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+def create_slice_in_scope():
+    df = ROOT.ROOT.RDataFrame(4).Define("x", "(double)rdfentry_")
+    npy = df.AsNumpy()
+    x = npy["x"]
+    x2 = x[:2]
+    return x2
+
+
+class RDataFrameAsNumpy(unittest.TestCase):
+    def test_branch_dtypes(self):
```
Comment (etejedor) [inline bindings/pyroot_experimental/PyROOT/test/rdataframe_asnumpy.py]: Thank you @stwunsch , just one more thing: to be compliant with the rest of the pythonisation tests, can we add a bit of documentation to the test class to explain what the pythonisation is doing (and thus, what we are testing)? See e.g.
https://github.com/root-project/root/blob/master/bindings/pyroot_experimental/PyROOT/test/ttree_iterable.py#L10

Comment (stwunsch) [reply bindings/pyroot_experimental/PyROOT/test/rdataframe_asnumpy.py]: Done!

**Pull request merged**